### PR TITLE
Keep WEBMs and convert to WEBM

### LINKED ISSUE #1507: feat: Convert to WEBM, not MP3, if WEBM is compatible with FLEx
Currently, we convert large audio files to MP3. If WEBM files are compatible with FLEx, we should convert to WEBM with OPUS encoding instead. OPUS encodes audio files at both a higher quality and in smaller files than MP3.

First, we need to make sure WEBM files are compatible with FLEx.

### PR DESCRIPTION
**Pull request by laineyhm**
Fixes #1507 

## Description

Recently, it seems that the Geckofx .NET wrapper in FLEx has been updated to allow WEBM files. (Please see this issue for discussion of past difficulties with WEBM audio in FLEx: [](https://jira.sil.org/browse/LT-20864).)

So, instead of converting recordings and uploads to MP3 or WAV format, we can now keep recordings in WEBM and convert formats like OGG, FLAC, and M4A to WEBM for use with FLEx. With WEBM in opus codec, we can maintain higher audio quality and smaller file sizes for audio than with MP3. 

Note: although in one commit, I tested with aflt codec, we are going to continue to use opus codec.

### Type of Change

Only keep lines below that describe this change, then delete the rest.

- New feature (non-breaking change which adds functionality)

(This is really a change to the audio features, not a new feature. We hope, however, that it will enble users to do more with the audio they save, since it will be of better quality).

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have reviewed the title/description of this PR which will be used as the squashed PR commit message
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works

## How to test

Here is a folder of test files: 
[Audio Test Upload Files.zip](https://github.com/sillsdev/web-languageforge/files/9730261/Audio.Test.Upload.Files.zip)

- [ ] Record some audio in Language Forge. Save it, then click the three dots next to the sound player. Examine the file name. It should be a WEBM file still (not converted to MP3 or WAV).
- [ ] Upload OGG, FLAC, and M4A files. Click the three dots next to the sound player and examine the file names. They should be WEBM files.
- [ ] Upload an MP3 file. Click the three dots next to the sound player and examine the file name. It should be an MP3 file still (=not converted. This is the same as we were doing before.).
- [ ] Upload a WAV file. Click the three dots next to the sound player and examine the file name. It should be a WAV file still (=not converted. This is the same as we were doing before.).

## qa.languageforge.org testing

Testers: Check the box and put in a date/time to sign-off/attest the feature works as expected on qa.languageforge.org

- [ ] Tester1 (YYYY-MM-DD HH:MM)
- [ ] Tester2 (YYYY-MM-DD HH:MM)

### CODE REVIEW

File: src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php [hunk 1]
```diff
@@ -103,17 +103,16 @@ public static function uploadAudioFile($projectId, $mediaType, $tmpFilePath)
             // convert audio file to mp3 or wav format if necessary
             // FLEx only supports mp3 or wav format as of 2022-09
```
Comment (megahirt) [inline src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php:104]:
```suggestion
```

File: src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php [hunk 2]
```diff
@@ -103,17 +103,16 @@ public static function uploadAudioFile($projectId, $mediaType, $tmpFilePath)
...
 
-            if (strcmp(strtolower($fileExt), ".mp3") !== 0 && strcmp(strtolower($fileExt), ".wav") !== 0) {
+            if (strcmp(strtolower($fileExt), ".mp3") !== 0 && strcmp(strtolower($fileExt), ".wav") !== 0 && strcmp(strtolower($fileExt), ".webm") !== 0) {
                 //First, find the duration of the file
                 $ffprobeCommand = `ffprobe -i $tmpFilePath -show_entries format=duration -v quiet -of csv="p=0" 2> /dev/null`;
                 $audioDuration = floatval($ffprobeCommand);
 
                 // Convert to .wav if the result will be less than 1 MB (recording is shorter than 5.6 seconds)
                 // and .mp3 otherwise (recording is longer than 5.6 seconds)
```
Comment (megahirt) [inline src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php:112]:
```suggestion
```

**Review by megahirt — COMMENTED** [review]
Let's remove these old comments.  Nice work!  I'll approve once the outdated comments are gone.  We've really gone full circle with the new knowledge we have about webm and opus!

**Review by megahirt — APPROVED** [review]
This is the direction we want to move it - thanks @laineyhm !

**Comment by myieye** [comment]
@laineyhm Everything's working fine for me in QA except for uploading the file types that get converted to WEBM.
When I upload them I can't play them. When I download them again they're 0kb. So it seems something in the conversion/saving process isn't working quite right.

https://user-images.githubusercontent.com/12587509/197146062-d691ae8a-0f77-491b-b70e-1638382a0497.mp4

**Comment by megahirt** [comment]
@myieye thank you for doing this thorough testing!  @laineyhm can sort out the bug on Tuesday when she's back in the office.

**Pull request merged**